src/components/CheckboxFilterContainer.ts

```diff
@@ -0,0 +1,154 @@
+import { Component, ReactElement, createElement } from "react";
+import { findDOMNode } from "react-dom";
+
+import { Alert } from "./Alert";
+import { CheckboxFilter, CheckboxFilterProps } from "./CheckboxFilter";
+import { Utils, parseStyle } from "../utils/ContainerUtils";
+
+import * as classNames from "classnames";
+import * as dijitRegistry from "dijit/registry";
+import * as dojoConnect from "dojo/_base/connect";
+
+interface WrapperProps {
+    class: string;
+    style: string;
+    friendlyId: string;
+    mxform?: mxui.lib.form._FormBase;
+}
+
+export interface ContainerProps extends WrapperProps {
+    listviewEntity: string;
+    caption: string;
+    filterBy: filterOptions;
+    attribute: string;
+    attributeValue: string;
+    constraint: string;
+    defaultChecked: boolean;
+    unCheckedFilterBy: filterOptions;
+    unCheckedAttribute: string;
+    unCheckedAttributeValue: string;
+    unCheckedConstraint: string;
+}
+
+
+export type filterOptions = "attribute" | "XPath" | "None";
+type HybridConstraint = Array<{ attribute: string; operator: string; value: string; path?: string; }>;
+
+export interface ListView extends mxui.widget._WidgetBase {
+    _datasource: {
+        _constraints: HybridConstraint | string;
+    };
+    filter: {
+        [key: string ]: HybridConstraint | string;
+    };
+    update: () => void;
+    _entity: string;
+}
+
+export interface ContainerState {
+    listviewAvailable: boolean;
+    targetListView?: ListView;
+    targetNode?: HTMLElement;
+    validationPassed?: boolean;
+}
+
+export default class CheckboxFilterContainer extends Component<ContainerProps, ContainerState> {
+
+    constructor(props: ContainerProps) {
+        super(props);
+
+        this.state = { listviewAvailable: true };
+        this.handleChange = this.handleChange.bind(this);
+        this.connectToListView = this.connectToListView.bind(this);
+        // Ensures that the listView is connected so the widget doesn't break in mobile due to unpredictable render time
+        dojoConnect.connect(props.mxform, "onNavigation", this, this.connectToListView);
```

Comment:
Here... `this.connectToListView.bind(this)`

Reply:
Was removed long time